apps/worker/src/app/workflow/usecases/run-job/run-job.usecase.ts

```diff
@@ -114,26 +117,81 @@ export class RunJob {
       throw error;
     } finally {
       if (shouldQueueNextJob) {
-        const newJob = await this.queueNextJob.execute(
-          QueueNextJobCommand.create({
-            parentId: job._id,
-            environmentId: job._environmentId,
-            organizationId: job._organizationId,
-            userId: job._userId,
-          })
-        );
-
-        // Only remove the attachments if that is the last job
-        if (!newJob) {
-          await this.storageHelperService.deleteAttachments(job.payload?.attachments);
-        }
+        await this.tryQueueNextJobs(job);
       } else {
         // Remove the attachments if the job should not be queued
         await this.storageHelperService.deleteAttachments(job.payload?.attachments);
       }
     }
   }
 
+  /**
+   * Attempts to queue subsequent jobs in the workflow chain.
+   * If queueNextJob.execute returns undefined, we stop the workflow.
+   * Otherwise, we continue trying to queue the next job in the chain.
+   */
+  private async tryQueueNextJobs(job: JobEntity): Promise<void> {
```

Comment:
this was needed because not we have a possible exception on finally block which is not handled. 

for the following workflow SMS => DELAY => IN-APP
the problem is the we handle the delay step (calculate the duration) while we executing the SMS job. 
a better solution would be to handle the DELAY step while we execute the DELAY job.
a refactor will take a lot of time and we will need to create backward compatibility support for old jobs as well.